$OpenBSD: patch-buildtools_wafadmin_Tools_ccroot_py,v 1.1 2015/07/18 02:11:13 jca Exp $

- at build time, encode the full library name + version in the SONAME

--- buildtools/wafadmin/Tools/ccroot.py.orig	Sun Jan 27 05:51:43 2013
+++ buildtools/wafadmin/Tools/ccroot.py	Fri Jan 29 08:43:05 2016
@@ -68,6 +68,7 @@ def get_cc_version(conf, cc, gcc=False, icc=False):
 			'__FreeBSD__' : 'freebsd',
 			'__NetBSD__'  : 'netbsd',
 			'__OpenBSD__' : 'openbsd',
+			'__Bitrig__'  : 'bitrig',
 			'__sun'       : 'sunos',
 			'__hpux'      : 'hpux',
 			'__sgi'       : 'irix',
@@ -588,7 +589,10 @@ def apply_vnum(self):
 		name2 = libname + '.' + nums[0]
 
 	if self.env.SONAME_ST:
-		v = self.env.SONAME_ST % name2
+		if sys.platform.startswith('openbsd') or sys.platform.startswith('bitrig'):
+			v = self.env.SONAME_ST % name3
+		else:
+			v = self.env.SONAME_ST % name2
 		self.env.append_value('LINKFLAGS', v.split())
 
 	bld = self.bld
